cmd/restic/cmd_description.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/restic/restic/internal/debug"
+	"github.com/restic/restic/internal/errors"
+	"github.com/restic/restic/internal/repository"
+	"github.com/restic/restic/internal/restic"
+	"github.com/restic/restic/internal/textfile"
+	"github.com/restic/restic/internal/ui"
+	"github.com/restic/restic/internal/ui/termstatus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func newDescriptionCommand() *cobra.Command {
+	var opts changeDescriptionOptions
+
+	cmd := &cobra.Command{
+		Use:   "description snapshotID [--set description | --set-file description]",
+		Short: "View or modify the description of snapshots",
+		Long: `
+The "description" command allows you to view or modify the description on an existing snapshot.
+
+The special snapshotID "latest" can be used to restore the latest snapshot in the
+repository.
+
+EXIT STATUS
+===========
+
+Exit status is 0 if the command was successful.
+Exit status is 1 if there was any error.
+Exit status is 10 if the repository does not exist.
+Exit status is 11 if the repository is already locked.
+Exit status is 12 if the password is incorrect.
+`,
+		GroupID:           cmdGroupDefault,
+		DisableAutoGenTag: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			term, cancel := setupTermstatus()
+			defer cancel()
+			return runDescription(cmd.Context(), opts, globalOptions, term, args)
+		},
+	}
+
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+type changeDescriptionOptions struct {
+	descriptionOptions
+	removeDescription bool
+}
+
+func (opts *changeDescriptionOptions) AddFlags(f *pflag.FlagSet) {
+	f.BoolVar(&opts.removeDescription, "remove-description", false, "remove the description from a snapshot")
+	opts.descriptionOptions.AddFlags(f)
+}
+
+func (opts *changeDescriptionOptions) Check() error {
+	err := opts.descriptionOptions.Check()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type descriptionOptions struct {
+	Description     string
+	DescriptionFile string
+}
+
+func (opts *descriptionOptions) AddFlags(f *pflag.FlagSet) {
+	f.StringVar(&opts.Description, "description", "", "set the description of this snapshot")
+	f.StringVar(&opts.DescriptionFile, "description-file", "", "set the description of this snapshot to the content of the file")
+}
+
+func (opts *descriptionOptions) Check() error {
+	if len(opts.Description) > 0 && len(opts.DescriptionFile) > 0 {
+		return errors.Fatal("--description and --description-file cannot be used together")
+	}
+
+	return nil
+}
+
+// readDescription returns the description text specified by either the
+// `--description` option or the content of the `--description-file`
+func readDescription(opts descriptionOptions) (string, error) {
+	descriptionScanner := bufio.NewScanner(strings.NewReader(opts.Description))
```

Comment:
What is the bufio construction used for when `opts.Description != ""`? The function could just return the `opts.Description`.